### PR TITLE
please.sh: teach `create_sdk_artifact` to use `architecture` instead of `bitness`

### PR DESCRIPTION
**Pull request by dennisameling**
Git for Windows's toolchain is moving away from "bitness" as it doesn't support arm64. See the attached discussion for more details.

Ref: https://github.com/git-for-windows/git/discussions/4203

### CODE REVIEW

**Comment by dennisameling** [comment]
The failing `check-for-missing-dlls` jobs are unrelated to this PR, but rather to yesterday's commit to `git-sdk-64`: https://github.com/git-for-windows/git-sdk-64/commit/b980ccf168d918d6262398ed99f2731998245562

File: please.sh [hunk 1]
```diff
@@ -4470,11 +4470,12 @@ publish_sdk () { #
...
 }
 
-create_sdk_artifact () { # [--out=<directory>] [--git-sdk=<directory>] [--bitness=(32|64|auto)] [--force] <name>
+create_sdk_artifact () { # [--out=<directory>] [--git-sdk=<directory>] [--architecture=(amd64|x86|arm64)] [--bitness=(32|64)] [--force] <name>
 	git_sdk_path=/
 	output_path=
 	force=
-	bitness=auto
+	architecture=
+	bitness=
```
Comment (dennisameling) [inline please.sh:4478]: I'm not seeing any places that are relying on `bitness=auto` (including `setup-git-for-windows-sdk`), so I think we're good to explicitly require `--bitness` or `--architecture` to be set. Please let me know if that's not the case.

**Comment by rimrul** [comment]
> The failing `check-for-missing-dlls` jobs are unrelated to this PR, but rather to yesterday's commit to `git-sdk-64`: [git-for-windows/git-sdk-64@b980ccf](https://github.com/git-for-windows/git-sdk-64/commit/b980ccf168d918d6262398ed99f2731998245562)

That issue was fixed in https://github.com/git-for-windows/git-sdk-64/commit/f0c95e95244200f44aae750c508a1d080bdfc88c.
I re-ran the failed checks, they're looking good now.

**Comment by dennisameling** [comment]
Now that https://github.com/git-for-windows/git-sdk-arm64/pull/6 has been merged, I can confirm that this also works as expected on `git-sdk-arm64`:

```
$ /usr/src/build-extra/please.sh create-sdk-artifact --architecture=aarch64 --out=/sdk-build-installers build-installers
Preparing worktree (detached HEAD 4f9bda7791)
Updating index flags: 100% (89057/89057), done.
Updating files: 100% (993/993), done.
grep: /clangarm64/bin/git.exe: No such file or directory
Updating index flags: 100% (89057/89057), done.
Updating files: 100% (16948/16948), done.
Output written to 'C:/git-sdk-arm64/sdk-build-installers'
```

Small note: the `no such file or directory` warning already pops up on this script today without the changes in this PR:

```
$ /usr/src/build-extra/please.sh create-sdk-artifact --bitness=64 --out=/sdk-build-installers build-installers
Preparing worktree (detached HEAD 71754d7913)
Updating index flags: 100% (118700/118700), done.
Updating files: 100% (1201/1201), done.
grep: /mingw64/bin/git.exe: No such file or directory
Updating index flags: 100% (118700/118700), done.
Updating files: 100% (16094/16094), done.
Output written to 'C:/git-sdk-64/sdk-build-installers'
```

CC @dscho

**Comment by dscho** [comment]
> Small note: the `no such file or directory` warning already pops up on this script today without the changes in this PR:

I believe that the problem is solved via 50deb2bca5dc8ef16ee67846fe4da0da3faf1c35 for i686, and may need something similar for x86_64.

File: please.sh (hunk 1 above)
Comment (dscho) [reply please.sh:3099]: I thought my manual runs were relying on this, allowing me to be lazy and not specify anything but the `<name>` ;-) But I guess I broke it long ago. So we're good to leave `--bitness=auto` behind.

**Review by dscho — APPROVED** [review]

**Pull request merged**